RTClib.h

```diff
@@ -271,14 +84,16 @@ enum Ds1307SqwPinMode {
     @brief  RTC based on the DS1307 chip connected via I2C and the Wire library
 */
 /**************************************************************************/
-class RTC_DS1307 {
+class RTC_DS1307 : RTC {
 public:
-  boolean begin(void);
-  static void adjust(const DateTime &dt);
-  uint8_t isrunning(void);
-  static DateTime now();
-  static Ds1307SqwPinMode readSqwPinMode();
-  static void writeSqwPinMode(Ds1307SqwPinMode mode);
+  boolean begin(const DateTime &dt = DateTime::COMPILE_DT);
```

Comment:
Most RTC modules can hold a small battery that enables the RTC to keep the time even when the rest of the circuit is unpowered. This is why you generally do **not** want to set the RTC every time the sketch starts. With this implementation of `begin()`, it would become impossible to not set the time at startup, which would make the backup battery useless.

Reply:
Fixed by writing two `begin()` methods for each concrete RTC class: one for using the RTC's current date/time and another for setting the date/time.

Reply:
I tend to agree with @edgar-bonet. I also see little value in second begin.

Reply:
@daveythacher I wrote two `being()` methods as the `RTC_Millis` and `RTC_Micros` classes did not support a `begin()` method without a `DateTime` parameter whereas all the other RTC classes only had a `being()` method with no parameters. Writing both `begin()` methods ensures that other programmers don't have to worry about which RTC is being used when using `begin()`.